feg/gateway/services/session_proxy/credit_control/gy/gy_client.go

```diff
@@ -447,6 +447,9 @@ func getReceivedCredits(cca *CCADiameterMessage) []*ReceivedCredits {
 			if mscc.FinalUnitIndication.Action == Redirect {
 				receivedCredits.RedirectServer = mscc.FinalUnitIndication.RedirectServer
 			}
+			if mscc.FinalUnitIndication.Action == RestrictAccess {
```

Comment:
Related to the previous comment, once refactored, here we will just have to do something like 

```
receivedCredits = False
if mscc.FinalUnitIndication != nil {
	receivedCredits.isFinalUnitIdicationValid = True
	receivedCredits.FinalUnitIndication = mscc.FinalUnitIndication
}
```
And if we ever change FinalUnitIndication message, we won't need to modify the feg at all